src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -691,7 +690,7 @@ private static int compareImports(String import1, String import2) {
         final String separator = "\\.";
         final String[] import1Tokens = import1.split(separator);
         final String[] import2Tokens = import2.split(separator);
-        for (int i = 0; i < import1Tokens.length && i != import2Tokens.length; i++) {
+        for (int i = 0; i != import1Tokens.length && i != import2Tokens.length; i++) {
```

Comment:
If we allow this, then imports with un-even lengths will be an issue, right?

One example is `java.util.Map` and `java.util.Map.Entry`.
The 2 are equal in every position, except one has `.Entry` on the end, which makes it have 1 greater length then the other.
The method will see the 2 imports as completely equal which is incorrect.

Can we add something after the `for` loop like:
````
if (result == 0 && import1Tokens.length != import2Tokens.length) {
    result = Integer.compare(import1Tokens.length, import2Tokens.length);
}
````

Reply:
Example case:
````
$ cat TestClass.java
import java.util.Map.Entry;
import java.util.Map;

import com.B;
import com.A;

public class TestClass {
    void method() {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="CustomImportOrderCheck">
<property name="sortImportsInGroupAlphabetically" value="true" />
<property name="specialImportsRegExp" value="com" />
<property name="customImportOrderRules" value="STANDARD_JAVA_PACKAGE###SPECIAL_IMPORTS" />
</module>
    </module>
</module>

$ java -jar checkstyle-8.5-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:5: Wrong lexicographical order for 'com.A' import. Should be before 'com.B'. [CustomImportOrder]
Audit done.
Checkstyle ends with 1 errors.
````

No violation on `java.util.Map.Entry`.

Reply:
Done.